### PR TITLE
[video] Add missing player properties

### PR DESCRIPTION
**Pull request by behenate**
# Why

`expo-video` was still missing some of the basic properties that `expo-av` had. This PR adds them: 

- `isLooping`
- `rate`
- `shouldCorrectPitch` - this is iOS and Android exclusive. The web player always corrects the pitch, and has no configuration options
- `positionMillis` - clamping is has implementation only on iOS, on other platforms the default clamping behaviour is ok

I kept the same property names as `expo-av`, each property is a separate commit so hopefully it's a bit easier to review

# How

Added the properties and some controls to test them to the `VideoScreen`. Also did some minor updates to the docs, such as adding the missing defaults. 

# Test Plan

Tested in BareExpo on Android, iOS, web

### CODE REVIEW

File: packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt [hunk 1]
```diff
@@ -182,6 +184,35 @@ class VideoModule : Module() {
             ref.player.currentPosition
           }
         }
+        .set { ref: VideoPlayer, position: Long ->
+          appContext.mainQueue.launch {
+            ref.player.seekTo(position)
+          }
+          appContext.mainQueue.launch {
+            ref.player.currentPosition.toInt()
+          }
```
Comment (lukmccall) [inline packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt:193]:
```suggestion
```
Why do we have those lines? I think we can remove them, but correct me if I'm wrong.

File: packages/expo-video/ios/VideoModule.swift [hunk 2]
```diff
@@ -118,6 +118,45 @@ public final class VideoModule: Module {
         player.staysActiveInBackground = staysActive
       }
 
+      Property("isLooping") { (player: VideoPlayer) -> Bool in
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoModule.swift:121]:
```suggestion
      Property("isLooping") { (player) -> Bool in
```
We can drop the type here

File: packages/expo-video/ios/VideoModule.swift [hunk 3]
```diff
@@ -118,6 +118,45 @@ public final class VideoModule: Module {
...
+        return player.isLooping
+      }
+      .set { (player, isLooping: Bool) in
+        player.isLooping = isLooping
+      }
+
+      Property("positionMillis") {(player: VideoPlayer) -> Int64 in
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoModule.swift:128]:
```suggestion
      Property("positionMillis") {(player) -> Int64 in
```

File: packages/expo-video/ios/VideoModule.swift [hunk 4]
```diff
@@ -118,6 +118,45 @@ public final class VideoModule: Module {
...
+        return Int64(time.seconds * 1000.0)
+      }
+      .set { (player, positionMillis: Int64) in
+        // Only clamp the lower limit, AVPlayer automatically clamps the upper limit.
+        let clampedMillis = max(0, positionMillis)
+        let timeToSeek = CMTimeMakeWithSeconds(Float64(clampedMillis) / 1000.0, preferredTimescale: .max)
+        player.pointer.seek(to: timeToSeek, toleranceBefore: .zero, toleranceAfter: .zero)
+      }
+
+      Property("rate") { (player: VideoPlayer) -> Float in
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoModule.swift:139]:
```suggestion
      Property("rate") { (player) -> Float in
```

File: packages/expo-video/ios/VideoModule.swift [hunk 5]
```diff
@@ -118,6 +118,45 @@ public final class VideoModule: Module {
...
+        return player.desiredRate
+      }
+      .set { (player, rate: Float) in
+        player.desiredRate = rate
+      }
+
+      Property("shouldCorrectPitch") { (player: VideoPlayer) -> Bool in
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoModule.swift:146]:
```suggestion
      Property("shouldCorrectPitch") { (player) -> Bool in
```

File: packages/expo-video/ios/VideoModule.swift [hunk 6]
```diff
@@ -118,6 +118,45 @@ public final class VideoModule: Module {
...
+        return player.shouldCorrectPitch
+      }
+      .set { (player, shouldCorrectPitch: Bool) in
+        player.shouldCorrectPitch = shouldCorrectPitch
+      }
+
+      Property("volume") { (player: VideoPlayer) -> Float in
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoModule.swift:153]:
```suggestion
      Property("volume") { (player) -> Float in
```

File: packages/expo-video/ios/VideoPlayer.swift [hunk 7]
```diff
@@ -1,11 +1,29 @@
...
 import AVFoundation
+import MediaPlayer
 import ExpoModulesCore
 
 internal final class VideoPlayer: SharedRef<AVPlayer>, Hashable {
   lazy var contentKeyManager = ContentKeyManager()
 
+  var isLooping = false {
+    didSet {
+      self.applyIsLooping()
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoPlayer.swift:12]:
```suggestion
     applyIsLooping()
```
Can we remove the `self` from here?

File: packages/expo-video/ios/VideoPlayer.swift [hunk 8]
```diff
@@ -16,17 +34,80 @@ internal final class VideoPlayer: SharedRef<AVPlayer>, Hashable {
...
+    playerItemObserver = nil
+
+    if let currentItem = pointer.currentItem, isLooping {
+      playerItemObserver = NotificationCenter.default.addObserver(
+        forName: NSNotification.Name.AVPlayerItemDidPlayToEndTime,
+        object: pointer.currentItem,
+        queue: nil) { [weak self] _ in
+          self?.pointer.seek(to: .zero)
+          self?.pointer.play()
+      }
```
Comment (lukmccall) [inline packages/expo-video/ios/VideoPlayer.swift:107]:
```suggestion
      playerItemObserver = NotificationCenter.default.addObserver(
        forName: NSNotification.Name.AVPlayerItemDidPlayToEndTime,
        object: pointer.currentItem,
        queue: nil
      ) { [weak self] _ in
        self?.pointer.seek(to: .zero)
        self?.pointer.play()
      }
```

**Review by lukmccall — APPROVED** [review]

**Comment by JustJoostNL** [comment]
Any updates on a canary release for expo-video? 👀

**Review by amandeepmittal — APPROVED** [review]

File: packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt [hunk 9]
```diff
@@ -182,6 +184,32 @@ class VideoModule : Module() {
             ref.player.currentPosition
           }
         }
+        .set { ref: VideoPlayer, position: Long ->
+          appContext.mainQueue.launch {
+            ref.player.seekTo(position)
+          }
+        }
+
+      Property("rate")
```
Comment (tsapeta) [inline packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt:193]: Maybe `playbackRate` as on Web https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/playbackRate ?

File: packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt [hunk 10]
```diff
@@ -173,7 +175,7 @@ class VideoModule : Module() {
           }
         }
 
-      Property("currentTime")
+      Property("positionMillis")
```
[the remark this answers is not in the document]
Comment (lukmccall) [reply packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt:178]: Let's rename it to the `currentTime` 😄

**Review by tsapeta — APPROVED** [review]

**Pull request merged**